### ISSUE TITLE
entered unreachable code at script/dom/globalscope.rs:505

### ISSUE THREAD

**Issue by mateon1** [servo/servo#18467]
Found on http://idokep.hu, but not reproduced

Stack trace (release mode):
```gdb
internal error: entered unreachable code (thread ScriptThread PipelineId { namespace_id: PipelineNamespaceId(1), index: PipelineIndex(4) }, at /shared/dev/rust/servo/components/script/dom/globalscope.rs:505)
stack backtrace:
   0:     0x55dbc6cb70dc - backtrace::backtrace::trace::h0f1eee7f17c620ad
   1:     0x55dbc6cb7112 - backtrace::capture::Backtrace::new::h1abdde591da67f3d
   2:     0x55dbc5a16865 - servo::main::{{closure}}::h2775028620a930e9
   3:     0x55dbc81cbb26 - std::panicking::rust_panic_with_hook
                        at /checkout/src/libstd/panicking.rs:612
   4:     0x55dbc60d707a - std::panicking::begin_panic::h99ca1191232fecb5
   5:     0x55dbc666592f - script::dom::globalscope::GlobalScope::enqueue_microtask::ha3fde2dee804af92
   6:     0x55dbc60df567 - std::panicking::try::do_call::h0351daa36c649fa3
   7:     0x55dbc81d2bcc - panic_unwind::__rust_maybe_catch_panic
                        at /checkout/src/libpanic_unwind/lib.rs:99
   8:     0x55dbc6810ff2 - script::script_runtime::enqueue_job::hf76cd0077641b9bd
   9:     0x55dbc7582b42 - _ZN9JSRuntime17enqueuePromiseJobEP9JSContextN2JS6HandleIP10JSFunctionEENS3_IP8JSObjectEE
                        at /shared/dev/rust/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/834ce35/mozjs/js/src/vm/Runtime.cpp:779
  10:     0x55dbc75bdc43 - intrinsic_EnqueuePromiseReactionJob
                        at /shared/dev/rust/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/834ce35/mozjs/js/src/vm/SelfHosting.cpp:1828
  11:     0x55dbc75333a1 - _ZN2js12CallJSNativeEP9JSContextPFbS1_jPN2JS5ValueEERKNS2_8CallArgsE
                        at /shared/dev/rust/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/834ce35/mozjs/js/src/jscntxtinlines.h:232
  12:     0x55dbc75017f9 - _ZN2js23InternalCallOrConstructEP9JSContextRKN2JS8CallArgsENS_14MaybeConstructE
                        at /shared/dev/rust/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/834ce35/mozjs/js/src/vm/Interpreter.cpp:453
  13:     0x55dbc7501b20 - InternalCall
                        at /shared/dev/rust/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/834ce35/mozjs/js/src/vm/Interpreter.cpp:498
  14:     0x55dbc7501b4a - _ZN2js13CallFromStackEP9JSContextRKN2JS8CallArgsE
                        at /shared/dev/rust/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/834ce35/mozjs/js/src/vm/Interpreter.cpp:504
  15:     0x55dbc750eb76 - Interpret
                        at /shared/dev/rust/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/834ce35/mozjs/js/src/vm/Interpreter.cpp:2873
  16:     0x55dbc750147b - _ZN2js9RunScriptEP9JSContextRNS_8RunStateE
                        at /shared/dev/rust/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/834ce35/mozjs/js/src/vm/Interpreter.cpp:399
  17:     0x55dbc75018c8 - _ZN2js23InternalCallOrConstructEP9JSContextRKN2JS8CallArgsENS_14MaybeConstructE
                        at /shared/dev/rust/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/834ce35/mozjs/js/src/vm/Interpreter.cpp:471
  18:     0x55dbc7501b20 - InternalCall
                        at /shared/dev/rust/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/834ce35/mozjs/js/src/vm/Interpreter.cpp:498
  19:     0x55dbc7501b4a - _ZN2js13CallFromStackEP9JSContextRKN2JS8CallArgsE
                        at /shared/dev/rust/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/834ce35/mozjs/js/src/vm/Interpreter.cpp:504
  20:     0x55dbc750eb76 - Interpret
                        at /shared/dev/rust/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/834ce35/mozjs/js/src/vm/Interpreter.cpp:2873
  21:     0x55dbc750147b - _ZN2js9RunScriptEP9JSContextRNS_8RunStateE
                        at /shared/dev/rust/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/834ce35/mozjs/js/src/vm/Interpreter.cpp:399
  22:     0x55dbc75018c8 - _ZN2js23InternalCallOrConstructEP9JSContextRKN2JS8CallArgsENS_14MaybeConstructE
                        at /shared/dev/rust/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/834ce35/mozjs/js/src/vm/Interpreter.cpp:471
  23:     0x55dbc7501b20 - InternalCall
                        at /shared/dev/rust/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/834ce35/mozjs/js/src/vm/Interpreter.cpp:498
  24:     0x55dbc7501b4a - _ZN2js13CallFromStackEP9JSContextRKN2JS8CallArgsE
                        at /shared/dev/rust/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/834ce35/mozjs/js/src/vm/Interpreter.cpp:504
  25:     0x55dbc750eb76 - Interpret
                        at /shared/dev/rust/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/834ce35/mozjs/js/src/vm/Interpreter.cpp:2873
  26:     0x55dbc750147b - _ZN2js9RunScriptEP9JSContextRNS_8RunStateE
                        at /shared/dev/rust/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/834ce35/mozjs/js/src/vm/Interpreter.cpp:399
  27:     0x55dbc75018c8 - _ZN2js23InternalCallOrConstructEP9JSContextRKN2JS8CallArgsENS_14MaybeConstructE
                        at /shared/dev/rust/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/834ce35/mozjs/js/src/vm/Interpreter.cpp:471
  28:     0x55dbc7501b20 - InternalCall
                        at /shared/dev/rust/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/834ce35/mozjs/js/src/vm/Interpreter.cpp:498
  29:     0x55dbc7501bb2 - _ZN2js4CallEP9JSContextN2JS6HandleINS2_5ValueEEES5_RKNS_13AnyInvokeArgsENS2_13MutableHandleIS4_EE
                        at /shared/dev/rust/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/834ce35/mozjs/js/src/vm/Interpreter.cpp:517
  30:     0x55dbc741a904 - _ZNK2js7Wrapper4callEP9JSContextN2JS6HandleIP8JSObjectEERKNS3_8CallArgsE
                        at /shared/dev/rust/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/834ce35/mozjs/js/src/proxy/Wrapper.cpp:165
  31:     0x55dbc7400dd0 - _ZNK2js23CrossCompartmentWrapper4callEP9JSContextN2JS6HandleIP8JSObjectEERKNS3_8CallArgsE
                        at /shared/dev/rust/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/834ce35/mozjs/js/src/proxy/CrossCompartmentWrapper.cpp:329
  32:     0x55dbc7410e3b - _ZN2js5Proxy4callEP9JSContextN2JS6HandleIP8JSObjectEERKNS3_8CallArgsE
                        at /shared/dev/rust/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/834ce35/mozjs/js/src/proxy/Proxy.cpp:401
  33:     0x55dbc74120bd - _ZN2js10proxy_CallEP9JSContextjPN2JS5ValueE
                        at /shared/dev/rust/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/834ce35/mozjs/js/src/proxy/Proxy.cpp:689
  34:     0x55dbc75333a1 - _ZN2js12CallJSNativeEP9JSContextPFbS1_jPN2JS5ValueEERKNS2_8CallArgsE
                        at /shared/dev/rust/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/834ce35/mozjs/js/src/jscntxtinlines.h:232
  35:     0x55dbc75016fd - _ZN2js23InternalCallOrConstructEP9JSContextRKN2JS8CallArgsENS_14MaybeConstructE
                        at /shared/dev/rust/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/834ce35/mozjs/js/src/vm/Interpreter.cpp:441
  36:     0x55dbc7501b20 - InternalCall
                        at /shared/dev/rust/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/834ce35/mozjs/js/src/vm/Interpreter.cpp:498
  37:     0x55dbc7501b4a - _ZN2js13CallFromStackEP9JSContextRKN2JS8CallArgsE
                        at /shared/dev/rust/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/834ce35/mozjs/js/src/vm/Interpreter.cpp:504
  38:     0x55dbc750eb76 - Interpret
                        at /shared/dev/rust/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/834ce35/mozjs/js/src/vm/Interpreter.cpp:2873
  39:     0x55dbc750147b - _ZN2js9RunScriptEP9JSContextRNS_8RunStateE
                        at /shared/dev/rust/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/834ce35/mozjs/js/src/vm/Interpreter.cpp:399
  40:     0x55dbc75018c8 - _ZN2js23InternalCallOrConstructEP9JSContextRKN2JS8CallArgsENS_14MaybeConstructE
                        at /shared/dev/rust/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/834ce35/mozjs/js/src/vm/Interpreter.cpp:471
  41:     0x55dbc7501b20 - InternalCall
                        at /shared/dev/rust/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/834ce35/mozjs/js/src/vm/Interpreter.cpp:498
  42:     0x55dbc7501bb2 - _ZN2js4CallEP9JSContextN2JS6HandleINS2_5ValueEEES5_RKNS_13AnyInvokeArgsENS2_13MutableHandleIS4_EE
                        at /shared/dev/rust/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/834ce35/mozjs/js/src/vm/Interpreter.cpp:517
  43:     0x55dbc726b228 - _Z20JS_CallFunctionValueP9JSContextN2JS6HandleIP8JSObjectEENS2_INS1_5ValueEEERKNS1_16HandleValueArrayENS1_13MutableHandleIS6_EE
                        at /shared/dev/rust/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/834ce35/mozjs/js/src/jsapi.cpp:2781
  44:     0x55dbc6981e44 - script::dom::bindings::codegen::Bindings::EventListenerBinding::EventListener::HandleEvent::h57e5cfa92d43cdf8
  45:     0x55dbc664fb82 - script::dom::event::invoke::had04956356d6f388
  46:     0x55dbc664eb45 - script::dom::event::Event::dispatch::h16f33f7683554430
  47:     0x55dbc66584c0 - script::dom::eventtarget::EventTarget::fire_event_with_params::ha10fa1180bf77453
  48:     0x55dbc6608652 - script::dom::document::Document::set_ready_state::h0b81eb0e52a7dbd2
  49:     0x55dbc6752587 - script::dom::servoparser::ServoParser::do_parse_sync::hb009e2043a779d99
  50:     0x55dbc6751f06 - script::dom::servoparser::ServoParser::parse_sync::h7aad80b3fe42953e
  51:     0x55dbc67503c0 - script::dom::servoparser::ServoParser::resume_with_pending_parsing_blocking_script::h4e4805c3a87f36a9
  52:     0x55dbc6611f5a - script::dom::document::Document::process_pending_parsing_blocking_script::h7f2ff43099e17a53
  53:     0x55dbc6611b31 - script::dom::document::Document::pending_parsing_blocking_script_loaded::he95d059f8d8aa446
  54:     0x55dbc66cb4fe - <script::dom::htmlscriptelement::ScriptContext as net_traits::FetchResponseListener>::process_response_eof::h1e774f544905b750
  55:     0x55dbc680f0db - <script::network_listener::ListenerRunnable<A, Listener> as script::script_thread::Runnable>::handler::h1ea5d021e9e008c6
  56:     0x55dbc6815351 - <script::script_thread::CancellableRunnable<T> as script::script_thread::Runnable>::main_thread_handler::h32c4de0836642461
  57:     0x55dbc683391c - script::script_thread::ScriptThread::handle_msg_from_script::h51bceeab9eb522ae
  58:     0x55dbc682c24e - script::script_thread::ScriptThread::handle_msgs::{{closure}}::hdd3fd9041a1efa8f
  59:     0x55dbc68288cf - script::script_thread::ScriptThread::handle_msgs::h4aadf07f4dff3bf3
  60:     0x55dbc6823ceb - script::script_thread::ScriptThread::start::h543e6bc16496788a
  61:     0x55dbc60d38f0 - std::sys_common::backtrace::__rust_begin_short_backtrace::hcbce24426485d7f2
  62:     0x55dbc6200605 - std::panicking::try::do_call::h8ef268d0617cbd63
  63:     0x55dbc81d2bcc - panic_unwind::__rust_maybe_catch_panic
                        at /checkout/src/libpanic_unwind/lib.rs:99
  64:     0x55dbc640a0b2 - <F as alloc::boxed::FnBox<A>>::call_box::h4e198348b3252fbb
  65:     0x55dbc81ca7ab - alloc::boxed::{{impl}}::call_once<(),()>
                        at /checkout/src/liballoc/boxed.rs:692
                         - std::sys_common::thread::start_thread
                        at /checkout/src/libstd/sys_common/thread.rs:21
                         - std::sys::imp::thread::{{impl}}::new::thread_start
                        at /checkout/src/libstd/sys/unix/thread.rs:84
  66:     0x7fae33649493 - start_thread
  67:     0x7fae33176abe - __clone
  68:                0x0 - <unknown>
Pipeline failed in hard-fail mode.  Crashing!
```

**Comment by asajeffrey** [comment]
IRC  chat at http://logs.glob.uno/?c=mozilla%23servo&s=12+Sep+2017&e=12+Sep+2017#c751150

TL;DR: there are still some switches on globals that aren't exhaustive.

**Comment by nox** [comment]
I am on this, I will just store the microtask queue in `GlobalScope`.

**Comment by nox** [comment]
Ugh, this cannot be done because of `DissimilarOriginWindow`. ://

**Comment by nox** [comment]
Actually it can be done, it's just one more garbage field in `DissimilarOriginWindow`, which doesn't matter.
